invenio_rdm_migrator/streams/actions/load/drafts.py

```diff
@@ -234,3 +242,169 @@ def _resolve_references(self, **kwargs):
+        # however, at DB level updating the bucket information would suffice
+        # the new record would point to it and the previous draft would be deleted.
+        assert self.data.record_bucket["locked"]
+        # delete bucket from STATE.BUCKET
+        # it is not linked to a draft anymore and records cannot get file updates
+        STATE.BUCKETS.delete(self.data.record_bucket["id"])
+        # need to update the bucket (e.g. for locking it)
+        yield Operation(OperationType.UPDATE, FilesBucket, self.data.record_bucket)
+
+        # TODO: impelement search on the state
+        files = STATE.FILE_RECORDS.search("record_id", self.data.draft["id"])
+        for file in files:
+            yield Operation(OperationType.DELETE, RDMDraftFile, file)
+            yield Operation(OperationType.INSERT, RDMRecordFile, file)
+
+    def _generate_draft_rows(self, **kwargs):
+        """Generates rows for a new draft."""
+        parent = self.data.parent
+        draft = self.data.draft
+        # delete draft
+        # can we simplify this by only passing an `id`?
+        # would mean all other fields are optional
+        # i.e. data validation is delegated to db exceptions
+        yield Operation(
+            OperationType.DELETE,
+            RDMDraftMetadata,
+            dict(
+                id=draft["id"],
+                json=draft["json"],
+                created=draft["created"],
+                updated=draft["updated"],
+                version_id=draft["version_id"],
+                index=draft["index"],
+                bucket_id=draft["bucket_id"],
+                parent_id=parent["id"],
+                expires_at=draft["expires_at"],
+                fork_version_id=draft["fork_version_id"],
+            ),
+        )
+
+        # create record
+        yield Operation(
+            OperationType.INSERT,
+            RDMRecordMetadata,
+            dict(
+                id=draft["id"],
+                json=draft["json"],  # what happens with fields such as $schema?
+                created=draft["updated"],  # creation of record ~ latest draft update
+                updated=draft["updated"],
+                version_id=1,
+                index=1,
+                bucket_id=draft["bucket_id"],
+                parent_id=parent["id"],
+            ),
+        )
+        # update parent
+        yield Operation(
+            OperationType.UPDATE,
+            RDMParentMetadata,
+            dict(
+                id=parent["id"],
+                json=parent["json"],
+                created=parent["created"],
+                updated=parent["updated"],
+                version_id=parent["version_id"],
+            ),
+        )
+
+        # update versioning
+        # not 100% sure this is correct
+        # if we publish a new draft, it would be
+        # not sure if next_draft_id=None always is the correct workflow
+        STATE.PARENTS.update(
```

Comment:
Not 100% sure this is correct...if we publish a new draft, it would be
not sure if next_draft_id=None always is the correct workflow